src/System.Runtime/tests/System/DecimalTests.cs

```diff
@@ -8,6 +8,7 @@
 
 namespace System.Tests
 {
+    [Collection("CultureTestCollection")]
```

Comment:
> I'd rather turn off xunit parallelism altogether than have RemoteExecutor become the go-to fix for every shared state problem

Neither RemoteExecutor nor disabling xunit parallelism should be necessary here.  Unless I'm missing something, changes to CurrentCulture (rather than DefaultThreadCurrentCulture) should only affect the current thread... as long as it's reset and doesn't wrap awaits or yields, it should be fine in the current process even with parallel test execution.